ovos_plugin_manager/templates/coreference.py

```diff
@@ -16,7 +16,7 @@ class CoreferenceSolverEngine:
         "vossos", "nossas", "vossas"]
 
     def __init__(self, config=None):
-        self.config = config or {}
+        self.config = config or get_plugin_config(section="coref")
```

Comment:
we do not want this in templates, the classes are meant to be usable outside mycroft context and should only accept config as arguments, if config integration is wanted the factory classes should be used instead

Reply:
Should this be removed from STT/TTS base classes then? This essentially is the same as `_init_config` in STT (https://github.com/OpenVoiceOS/OVOS-plugin-manager/blob/dev/ovos_plugin_manager/templates/stt.py#L59)

Reply:
No thats different, stt has a config_core property we need to keep for compat, it still accepts a dict in constructor

Reply:
Id love to drap that but we need to keep it because this class exists in mycroft core unlike most of the others

Reply:
I left the changes to templates that were already loading config and reverted the ones that defaulted to `{}`